pkg/signatures/cosign_sig_verifier.go

```diff
@@ -200,15 +200,18 @@ func (c *cosignSignatureVerifier) VerifySignature(ctx context.Context,
 	var mutex sync.Mutex
 	var wg sync.WaitGroup
 	for _, opts := range c.verifierOpts {
-		for _, sig := range sigs {
+		for cnt, sig := range sigs {
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
 				verifierRefs, err := verifyImageSignature(ctx, sig, hash, image, opts)
 				mutex.Lock()
 				defer mutex.Unlock()
 				if err != nil {
-					allVerifyErrs = multierror.Append(allVerifyErrs, err)
+					allVerifyErrs = multierror.Append(
+						allVerifyErrs,
+						errors.Wrapf(err, "verifying signature %d", cnt),
```

Comment:
Maybe it will be better to have a hash here? `sig.Digest()`

Reply:
The digest will be same for all signatures.